### ISSUE TITLE
change IP of patroni cluster nodes

### ISSUE THREAD

**Issue by ani-tvm** [zalando/patroni#1693]
We have 3 node patroni postgres cluster DB its working fine. We need to change the IP s of all 3 node and cluster VIP. How to achieve this safely without affecting the replication and all.

**Comment by CyberDem0n** [comment]
Maybe if you play around with using DNS names instead of IPs it would work, not sure if it worth the effort.

**Comment by ani-tvm** [comment]
thanks  CyberDem0n .

we are not using DNS. every hosts /etc/hosts having its name to IP alone (not other 2 nodes name to IP). 
So IP changing is not possible patroni clustered postgres?

**Comment by CyberDem0n** [comment]
If the primary changes its IP, Patroni will have to update `primary_conninfo` on replicas and restart postgres.

**Comment by ani-tvm** [comment]
so patroni or postgres or ETCD  configuration side do we need to do any changes?

**Comment by CyberDem0n** [comment]
It all depends on where and how you run Etcd.
If etcd nodes changing IPs you have to do it slowly, one-by-one, after carefully reading the Etcd [op-guide](https://github.com/etcd-io/etcd/tree/master/Documentation/op-guide).

**Comment by ani-tvm** [comment]
So patroni will not update the IP changes to the ETCD automatically?

**Comment by CyberDem0n** [comment]
I don't really get your question.
If you change `restapi.connect_address` and `postgresql.connect_address` and restart Patroni it will publish new values to DCS.

**Comment by ani-tvm** [comment]
thanks CyberDem0n

**Comment by nijesh8** [comment]
> If you change `restapi.connect_address` and `postgresql.connect_address`


Hi CyberDem0n,

Could you let me know how to do the below ? 
"> If you change `restapi.connect_address` and `postgresql.connect_address`
"


Any command or in which Config file I can change this ? 



Thanks, 
Nijesh

**Comment by 123vivekr** [comment]
@nijesh8 That is in the patroni.yaml configuration file

**Comment by nijesh8** [comment]
@123vivekr  / @CyberDem0n 

Thank you so much. 
It worked.

Here is what I did: 

In the local patroni.yml file in the POD (in which the IP changed), I updated the below to the new IP.
  restapi:
    connect_address: '10.y.xx.xx:8008'
  postgresql:
    connect_address: '10.y.xx.xx:5432'

And did the below:
patronictl restart <clustername> <pod name> 

This restarted the pod and updated the  IP correctly in the DCS.
Now the other pods can connect to this pod without issues. 

Thanks again, 
Nijesh
